### PR TITLE
misc-functions

### PR DESCRIPTION
**Pull request by HailSanta**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 9 functions (+0.12%) on `us`.

✅ This PR fixes 21 warnings!

File: src/evt/fx_api.c [hunk 1]
```diff
@@ -1,35 +1,121 @@
...
+            npc = resolve_npc(script, npcID);
+            if (npc == NULL) {
+                return ApiStatus_DONE2;
+            }
+            X = 0.0f;
+            Y = (npc->collisionHeight * 4) / 5;
+            Z = 0.0f;
+            R = npc->collisionHeight / 3;
+            break;
+
```
Comment (nanaian) [inline src/evt/fx_api.c:105]:
```suggestion
```

File: src/evt/fx_api.c [hunk 2]
```diff
@@ -1,35 +1,121 @@
...
+    switch (emoterType) {
+        case EMOTER_PLAYER:
+            // show emote from player
+            npc = (Npc*)-1;
+            X = 0.0f;
+            Y = (gPlayerStatus.colliderHeight * 2) / 3;
+            Z = 0.0f;
+            R = gPlayerStatus.colliderHeight / 3;
+            break;
+
```
Comment (nanaian) [inline src/evt/fx_api.c:93]:
```suggestion
```

File: src/evt/fx_api.c [hunk 3]
```diff
@@ -1,35 +1,121 @@
...
+    s32 duration = evt_get_variable(script, *args++);
+    s32 emoterType = evt_get_variable(script, *args++);
+    f32 dX = evt_get_float_variable(script, *args++);
+    f32 dY = evt_get_float_variable(script, *args++);
+    f32 dZ = evt_get_float_variable(script, *args++);
+    f32 radius = evt_get_float_variable(script, *args++);
+
+    Npc* npc;
+    s32 emoteHandle;
+    f32 X, Y, Z, R;
```
Comment (nanaian) [inline src/evt/fx_api.c:82]:
```suggestion
    f32 x, y, z, r;
```

File: src/evt/fx_api.c [hunk 4]
```diff
@@ -40,94 +126,186 @@ ApiStatus RemoveEffect(Evt* script, s32 isInitialCall) {
...
+    f32 pitch = evt_get_float_variable(script, *args++);
+    s32 emoterType = evt_get_variable(script, *args++);
+    f32 posX = evt_get_float_variable(script, *args++);
+    f32 posY = evt_get_float_variable(script, *args++);
+    f32 posZ = evt_get_float_variable(script, *args++);
+    f32 radius = evt_get_float_variable(script, *args++);
+    s32 outVar = *args++;
+
+    Npc* npc;
+    f32 X, Y, Z, R;
```
Comment (nanaian) [inline src/evt/fx_api.c:267]:
```suggestion
    f32 x, y, z, r;
```

File: src/evt/fx_api.c [hunk 5]
```diff
@@ -40,94 +126,186 @@ ApiStatus RemoveEffect(Evt* script, s32 isInitialCall) {
...
+    f32 pitch = evt_get_float_variable(script, *args++);
+    s32 emoterType = evt_get_variable(script, *args++);
+    f32 posX = evt_get_float_variable(script, *args++);
+    f32 posY = evt_get_float_variable(script, *args++);
+    f32 posZ = evt_get_float_variable(script, *args++);
+    f32 radius = evt_get_float_variable(script, *args++);
+    s32 duration = evt_get_variable(script, *args++);
+
+    Npc* npc;
+    f32 X, Y, Z, R;
```
Comment (nanaian) [inline src/evt/fx_api.c:224]:
```suggestion
    f32 x, y, z, r;
```

File: src/18C790.c [hunk 6]
```diff
@@ -267,7 +267,7 @@ void btl_state_update_celebration(void) {
                 btl_cam_use_preset(BTL_CAM_PRESET_C);
                 btl_cam_move(5);
                 dma_copy(_79EF40_ROM_START, _79EF40_ROM_END, _79EF40_VRAM);
-                sfx_play_sound(0x80000008);
+                sfx_play_sound(SOUND_80000008);
```
Comment (ethteck) [inline src/18C790.c:270]: I think 0x80000000 is a flag and there aren't unique sounds starting with that value... but I could be wrong. do you know anything about that?

File: src/cd180_len_38f0.c [hunk 7]
```diff
@@ -22,26 +22,191 @@ s32 D_8014C6F4[] = { 0x00000000, 0x00000000, 0x00000000, };
...
-
+Vtx D_8014E500[10] = {
```
Comment (ethteck) [inline src/cd180_len_38f0.c:25]: We should probably be extracting these with splat (see the vtx type)

File: src/evt/cam_api.c [hunk 8]
```diff
@@ -187,7 +187,70 @@ ApiStatus SetCamTarget(Evt* script, s32 isInitialCall) {
...
+        
+        data = heap_malloc(sizeof(*data));
+        script->userData = data;
+        cam = &gCameras[camID];
+        data->cam = cam;
+        data->useTarget = useTarget;
+        data->time = time;
+        
+        switch (data->useTarget) {
+        case 0:
```
Comment (ethteck) [inline src/evt/cam_api.c:218]: indent switch cases

File: src/evt/cam_api.c [hunk 9]
```diff
@@ -187,7 +187,70 @@ ApiStatus SetCamTarget(Evt* script, s32 isInitialCall) {
...
+            data->vel.y = (posY - cam->targetPos.y) / data->time;
+            data->vel.z = (posZ - cam->targetPos.z) / data->time;
+            break;
+        }
+    }
+    
+    data = script->userData;
+    cam = data->cam;
+    switch (data->useTarget) {
+    case 0:
```
Comment (ethteck) [inline src/evt/cam_api.c:234]: here2

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/18C790.c (hunk 6 above)
Comment (nanaian) [reply src/18C790.c:270]: 0x80000000 is for looping sounds I believe

Comment (HailSanta) [reply src/18C790.c:270]: Yes, there are also 0xB00000XX and I think one other. 0x80000008 is distinct from 0x8; it is NOT sound 8 looped. Giving them separate names is correct.

**Review by HailSanta — COMMENTED** [review]
effect organization

**Pull request merged**